src/univers/gem.py

```diff
@@ -242,6 +237,26 @@ def __gt__(self, other):
     def __ge__(self, other):
         return self.__cmp__(other) >= 0
 
+    @property
+    def segments(self):
```

Comment:
You should find a way such that segments is a list or tuple and not an iterable as this is always short and too much burden on the caller.... especially I would expect a property to return an actual collection not to yield things.
One thing could be to rename this (and the canonical one too) to `def _segments(self):` and have `a @property def segments(self): return tuple(self._segments())` ? this makes it simpler for the caller and does not break the existing "api" and behavious